src/hooks/useErc20.ts

```diff
@@ -59,7 +65,92 @@ export function useErc20(params: Params): Return {
           setIsLoading(false)
         })
     }
-  }, [address, dispatch, erc20, networkId])
+  }, [address, erc20, networkId, saveErc20s])
 
   return { isLoading, error, value: erc20 }
 }
+
+export type UseMultipleErc20Params = { addresses: string[]; networkId?: Network }
+
+/**
+ * Fetches multiple erc20 token details for given network and addresses
+ * More efficient method to fetch many tokens at once, and avoid unnecessary re-renders
+ *
+ * Tries to get it from globalState.
+ * If not found, tries to get it from the network.
+ * Saves to globalState if found.
+ *`value` is an object with the `address` as key and it's value is either `null` when not found or the erc20
+ * Returns `isLoading` to indicate whether fetching the value
+ * Returns `error` with the error messages, if any.
+ */
+export function useMultipleErc20(params: UseMultipleErc20Params): Return<Record<string, SingleErc20State>> {
+  const { addresses, networkId } = params
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState('')
+
+  const getMultipleErc20 = useGetMultipleErc20sFromGlobalState(networkId)
+  const saveErc20s = useSaveErc20sToGlobalState(networkId)
+
+  // local copy of globalState
+  const erc20s = useMemo(() => getMultipleErc20(addresses), [addresses, getMultipleErc20])
+  // check what on globalState has not been fetched yet
+  const toFetch = useMemo(() => addresses.filter((address) => !erc20s[address]), [addresses, erc20s])
+  // flow control
+  const running = useRef(false)
+
+  const loadFromNetwork = useCallback(
+    async (toFetch: string[]): Promise<void> => {
+      if (!networkId || toFetch.length === 0) {
+        return
+      }
+
+      running.current = true
+
+      setIsLoading(true)
+      setError('')
+
+      const promises = toFetch.map(
+        async (address): Promise<SingleErc20State> => {
+          try {
+            const erc20 = await getErc20Info({ tokenAddress: address, networkId, web3, erc20Api })
+
+            return erc20
+          } catch (e) {
+            const msg = `Failed to fetch erc20 details for ${address} on network ${networkId}`
+            console.error(msg, e)
+            setError((errors) => [errors, msg].filter(Boolean).join('\n'))
```

Comment:
should `errors` parameter be in singular?, after all it's call setError

` [errors, msg].filter(Boolean).join('\n')` is a bit geeky?, maybe we errors should be an array of strings instead of a string. What if you want to later show it in a <ul> for example. Maybe it should even be an object with the address as a key. This way, if your token is null, you know you have an error message waiting for you

Reply:
Good call, will make it an object

Reply:
haha, i saw my comment, and in the part I say 
` What if you want to later show it in a` there's a line break, cause I wrote an Html **UL** which was converted to HTML or sth messing with my comment